src/main/java/org/sagebionetworks/web/client/widget/entity/act/ApproveUserAccessModal.java

```diff
@@ -190,6 +193,58 @@ public void show() {
 	}
 	
 	@Override
+	public void onRevoke() {
+		if (userId == null) {
+			synAlert.showError(NO_USER_SELECTED);
+			return;
+		}
+		accessRequirement = view.getAccessRequirement();
+		view.setRevokeProcessing(true);
+		synapseClient.getEntityAccessApproval(datasetId, new AsyncCallback<PaginatedResults<AccessApproval>>() {
+
+			@Override
+			public void onFailure(Throwable caught) {
+				synAlert.handleException(caught);
+				view.setRevokeProcessing(false);
+			}
+
+			@Override
+			public void onSuccess(PaginatedResults<AccessApproval> result) {
+				List<AccessApproval> results = result.getResults();
+				Long accessReq = Long.parseLong(accessRequirement);
+				for (AccessApproval approval : results) {
+					if (approval.getAccessorId().equals(userId) && approval.getRequirementId().equals(accessReq)) {
```

Comment:
remove all access approvals for user in dataset (not just the selected access requirement).